utils/slice.go

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+// DeleteIndex moves the last element in the slice to index [i] and shrinks the
```

Comment:
I understand this is O(1) and not O(n) like the regular `slices.Delete` but it is less code and less complexity to just use the aforementioned builtin `slices.Delete` over adding a new method. 
`Deregister` isn't called yet in the code, and if called - it will be probably be called seldomly, so I'm not sure if it's worth adding it from a code maintenance perspective.

Reply:
This definitely isn't performance critical... But I also feel like I've re-implemented this same code like 5 times... So I feel like it is a reasonable helper to add